src/euphorie/deployment/commands/upgradedb.py

```diff
@@ -144,6 +154,7 @@ def main():
         add_group_id_to_session()
         add_published_to_session()
         add_last_modifier_id_to_session()
+        hash_passwords()
```

Comment:
@ale-rt Wow, looks like a great piece of work!
Just a suggestion: What if, instead of adding this as a general step to upgradedb, which could be run many times, you make an explicit upgrade step for this?
Then we would not need to worry about this comment, because the PW would only need to be encrypted once: either during the migration, or when the user sets / changes it. Does that make sense?

>         if BCRYPTED_PATTERN.match(password):
>             # The password is already encrypted, do not encrypt it again
>             # XXX this is broken with passwords that are actually an hash

Reply:
Nope, that has to stay because the events are fired when a row in the account table is updated, regardless of the column.
If you remove that, changing any column will encrypt the password again.

For me the `hash_passwords` method should live in the upgradedb script, but if you have a better place I can move it.

Reply:
ok, thanks for the explanation.
So, in theory, after `hash_passwords` has run once on the thousands of accounts, it will be quick the next time that updatedb is run again, since the pattern-matching will quickly detect that there is nothing to do?

Reply:
Yes